apps/aecore/src/aec_blocks.erl

```diff
@@ -162,133 +169,208 @@ new_key_with_state(LastBlock, CurrentKeyBlock, Miner, Trees0) ->
 
-    {UncompleteBlock#block{height = NewHeight, version = Version, miner = KeyToClaimLeader, key_hash = undefined}, Trees}.
+    {UncompleteBlock#block{height = NewHeight, version = Version, miner = KeyToClaimLeader, miner_hash = undefined}, Trees}.
 
 -spec to_header(block()) -> aec_headers:header().
-to_header(#block{height = Height,
-                 key_hash = PreviousKeyBlockHash,
-                 signature = LeaderSignature,
-                 prev_hash = PrevHash,
-                 txs_hash = TxsHash,
-                 root_hash = RootHash,
-                 target = Target,
-                 nonce = Nonce,
-                 time = Time,
-                 version = Version,
-                 pow_evidence = Evd,
-                 miner = Miner}) ->
+to_header(#block{} = B) ->
+    to_header(type(B), B).
+
+to_header(key, #block{height = Height,
+                      prev_hash = PrevHash,
+                      root_hash = RootHash,
+                      miner = Miner,
+                      target = Target,
+                      pow_evidence = Evd,
+                      nonce = Nonce,
+                      time = Time,
+                      version = Version}) ->
     #header{height = Height,
-            key_hash = PreviousKeyBlockHash,
-            signature = LeaderSignature,
             prev_hash = PrevHash,
-            txs_hash = TxsHash,
             root_hash = RootHash,
+            miner = Miner,
             target = Target,
+            pow_evidence = Evd,
             nonce = Nonce,
             time = Time,
-            pow_evidence = Evd,
-            version = Version,
-            miner = Miner}.
-
-from_header_and_txs(#header{height = Height,
-                            prev_hash = PrevHash,
-                            key_hash = PreviousKeyBlockHash,
-                            signature = LeaderSignature,
-                            txs_hash = TxsHash,
-                            root_hash = RootHash,
-                            target = Target,
-                            nonce = Nonce,
-                            time = Time,
-                            pow_evidence = Evd,
-                            version = Version,
-                            miner = Miner}, Txs) ->
+            version = Version};
+to_header(micro, #block{height = Height,
+                        prev_hash = PrevHash,
+                        root_hash = RootHash,
+                        txs_hash = TxsHash,
+                        miner_hash = MinerHash,
+                        signature = Signature,
+                        time = Time,
+                        version = Version}) ->
+    #header{height = Height,
+            prev_hash = PrevHash,
+            root_hash = RootHash,
+            txs_hash = TxsHash,
+            miner_hash = MinerHash,
+            signature = Signature,
+            time = Time,
+            version = Version}.
+
+from_header_and_txs(#header{} = H, Txs) ->
```

Comment:
This looks strange both `key` and `micro` type ignores the Txs? What would then be the point of this function?

Reply:
This is still WIP, `Txs` won't be ignored in microblocks.